deploy/factory.ts

```diff
@@ -0,0 +1,139 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types";
```

Comment:
This is a meta comment for all of these scripts. I'm not expecting these to be as heavily commented as the Solidity or Rust SDK -- this code is much more self-explanatory. That said, I'd like to see some high-level comments describing the different steps of the deployment.